### DISCUSSION TITLE
Cesta y favoritos

### DISCUSSION THREAD

**Discussion by GraceFinlay** [Telefonica/mistica-design#1017] [Proposals]
Buenas tardes,

Mi nombre es Grace y estoy trabajando en el proyecto de ecommerce global y os voy a indicar una serie de dudas relacionadas con Mística que han surgido en el desarrollo del proyecto.

# Contexto

El proyecto que está en marcha quiere incorporar unas mejoras en la tienda para que diferentes usuarios puedan comprar diferentes dispositivos. Estas mejoras que se quieren introducir son entre otras la mejora de los flujos de usuario y diferentes funcionalidades.

# Problema

Para este proyecto se requiere utilizar Mística y observamos que tenemos unas necesidades concretas en cuanto a componentes. Para nosotros es importante obtener feedback de vuestra parte con el fin de entender cómo podríamos implementar el proyecto usando Mística y qué cubra las necesidades planteadas.En este caso queremos un icono que tengas dos funcionalidades por un lado, indicar al usuario  que en la cesta hay productos (actualmente existe en Mística) y por otro lado, que cuando los usuarios añadan un producto a sus favoritos ese mismo icono con un corazón dentro, les indique que  el producto están en sus favoritos.

# Propuesta
![favoritos](https://user-images.githubusercontent.com/99174134/214295393-09d6c44a-056d-46e0-b671-86f4e978e342.png)

Para el diseño de este icono:

1. Hemos juntado dos iconos de Mística.
2. Se tendría que crear una variante mas para el icono de la compra donde no sólo indique que tenemos productos en la cesta sino que también hay favoritos añadidos.

Estamos abiertos a cualquier feedback que nos podáis dar para abordar el proyecto.

De antemano muchas gracias.

**Comment by yceballost** [comment]
Te cuento de nuevo los dos caminos que podríamos hacer aquí:
1. Incluirlo en Mística Icons, haciendo que exista un nuevo icono en el pack de iconos y así otros equipos puedan usarlo y sería muy fácil de usar en desarrollo, solo tendrían que poner el nombre del icono.
2. Icono custom para lo vuestro, en Mística sigue siendo posible tener iconos custom. El diseñador / diseñadora prepara el asset, exporta y el desarrollador lo importa en el código.

Lo ideal siempre es usar iconos del pack, pero hay ocasiones en que no merece la pena incluirlo dentro. 

Desventajas del punto 2. Si mañana se actualiza el pack de iconos de Movistar e incluye un icono de carrito "filled" a vosotros no se os actualizaría automáticamente. 

Dicho esto, personalmente me parece interesante la propuesta pero creo que podría confundirse con elementos de una "wishlist" o "favoritos" al usar el corazón dentro.

Por otro lado, los iconos no deben de tener 2 colores. A tu propuesta habría que eliminar el color rojo del corazón.

Me parece interesante añadir al pack de iconos un `shopping-cart` "filled" para poder hacer el juego entre vacío y lleno pero usaría otro visual para esta finalidad. Hiciste bench sobre cómo hacen esta interacción en otros ecommerce?

**Reply by GraceFinlay** [reply]
Buenas tardes, 

Lo que buscamos es esto:
<img width="124" alt="Icono Cesta y favoritos compartido" src="https://user-images.githubusercontent.com/99174134/214847223-cdf3ea80-3696-4f05-a5b0-9164852eab7d.png">
En la navegación la cesta de la compra y los favoritos estarían juntos, como puedes ver aquí:
![Captura de Pantalla 2023-01-26 a las 13 30 03](https://user-images.githubusercontent.com/99174134/214847716-4568d5f4-1a5f-47c0-9b86-05ba7a891fd1.png)

Entonces se pensó que el mismo icono sirviera de entrada para esta screen. Por lo tanto, cesta representaría a favoritos con el corazón ya sea vacío o lleno (en un sólo color como me has indicado antes) y el badge representaría el número de artículos que se encuentran en el carrito.

Podría hacerse con un pack de iconos  como indicas anteriormente. 

Quedamos a la espera de tu respuesta. Muchas gracias.

**Reply by AnaMontes11** [reply]
Buenos días Grace, las opciones que planteas del carrito con el corazon filled funcionan muy bien, pero creo que el que va a línea dará problemas de visibilidad al llevar un grosor tan fino. Otra opción que se me ocurre sería dejar ese que funciona bien para regular y el filled hacerlo de la siguiente forma:
![icon2](https://user-images.githubusercontent.com/88187691/215043751-a75df593-5974-4cbc-a835-139df1ed3547.jpg)
Espero que te ayude

**Comment by yceballost** [comment]
Cerramos por inactividad
